src/coreclr/src/vm/diagnosticserver.cpp

```diff
@@ -69,8 +70,20 @@ DWORD WINAPI DiagnosticServer::DiagnosticsServerThread(LPVOID)
                 continue;
             }
 
+            // check to see if command is enabled yet.
+            if (!IsCommandEnabled(message.GetHeader().CommandSet, message.GetHeader().CommandId))
+            {
+                DiagnosticsIpc::IpcMessage::SendErrorMessage(pStream, E_FAIL);
```

Comment:
Let's find (or define) a more specific error code.

Reply:
I had a new one defined, but didn't use it to avoid adding too much to the protocol.  If I recall the diagnostics range of error codes is running short--I believe there are only 1 or 2 left.  I'll see if I can't find a preexisting HRESULT code that makes sense here and create one if I can't.

Reply:
If defining more error code isn't an option, would `E_UNKNOWN_COMMAND` be better for this particular case? We're trying to notify that the runtime is not able to take a certain command, whereas E_FAIL could mean literally anything.

Reply:
Given that the error case only applies to the profiler the error code can be returned in the [payload](https://github.com/dotnet/diagnostics/blob/master/documentation/design-docs/ipc-protocol.md#returns-as-an-ipc-message-payload-3) of that response rather than as an agnostic mechanism.

Reply:
Are you proposing that I use a profiler error code for this?  Right now it returns unknown error, which feels too ambiguous.